src/util.rs

```diff
@@ -5,7 +5,9 @@ use std::{io, iter::repeat_with};
 
 use crate::error::IoResultExt;
 
-fn tmpname(prefix: &OsStr, suffix: &OsStr, rand_len: usize) -> OsString {
+// Hide this function in the docs as it's only exposed for usage in integration tests.
+#[doc(hidden)]
+pub fn tmpname(prefix: &OsStr, suffix: &OsStr, rand_len: usize) -> OsString {
```

Comment:
Instead of exposing this function, I'd just create a temporary directory inside a temporary directory. I.e.:

1. `let tmp = TempDir:new()`.
2. `TempDir::new_at(tmp.path().join("my_name")`